### PR TITLE
Add configmap in jenkins namespace for openshift-login plugin

### PR DESCRIPTION
**Pull request by waveywaves**
This will add a new configmap in jenkins namespace for
openshift login plugin.

Previously there were default roles in openshift-login plugin
and they can't be configured. Because of that we were using
a fork of upstream login plugin with roles configuration
according to our use case.

Now, a patch has been added in upstream login plugin through
which we can configure roles by providing a configmap

This patch will add the same configmap which will be used
by openshift-login plugin to configure roles.
We will deprecate our fork of login plugin and will move to
upstream version.

From image, we have removed our fork binary and used upstream
using this patch
fabric8io/openshift-jenkins-s2i-config#234

This will fix openshiftio/openshift.io#4608

### CODE REVIEW

**Comment by waveywaves** [comment]
@piyush-garg Do check this and let me know

**Comment by piyush-garg** [comment]
[test]

**Comment by piyush-garg** [comment]
@waveywaves You forgot to change the Jenkins snapshot version

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=f22a6f0a1365ddb3a7acdc0f365fe16a23986c7f) and click the update tenant button

**Comment by piyush-garg** [comment]
[test]

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=9241b9e26e01075759e3ac330d055e7e18f9a242) and click the update tenant button

**Comment by piyush-garg** [comment]
[test]

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=8a2eaad32679ca4ee9fa4f6ed889d2716d4378d7) and click the update tenant button

**Comment by piyush-garg** [comment]
[test]

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=4f9e21fb0aecb36ebd0a8298c4d01c7c11c1d7a6) and click the update tenant button

**Comment by piyush-garg** [comment]
[test]

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=ee04c7404f0c1f738bd3d05524d0eb2cbccdea39) and click the update tenant button

**Comment by piyush-garg** [comment]
[test]

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=2a5d2657f8624051655d9d7b53d0c75878744124) and click the update tenant button

File: environment/templates/fabric8-tenant-jenkins.yml [hunk 1]
```diff
@@ -834,11 +834,44 @@ objects:
...
+    Run-Update: admin,edit,view
+    View-Configure: admin
+    View-Create: admin
+    View-Delete: admin
+    Credentials-Create: admin
+    Credentials-Update: admin
+    Credentials-Delete: admin
+    Credentials-ManageDomains: admin
+    Credentials-View: admin,edit,view
+    SCM-Tag: admin,edit,view
```
Comment (piyush-garg) [inline environment/templates/fabric8-tenant-jenkins.yml]: Can you please add 
```
Slave-Configure: admin
Slave-Delete: admin
```

File: environment/templates/fabric8-tenant-jenkins.yml [hunk 2]
```diff
@@ -834,11 +834,44 @@ objects:
...
+    Overall-Administer: admin
+    Job-Build: admin,edit,view
+    Job-Configure: admin,edit,view
+    Job-Create: admin,edit,view
+    Job-Delete: admin,edit,view
+    Job-Cancel: admin,edit,view
+    Job-Wokspace: admin,edit,view
+    Job-Read: admin,edit,view
+    Job-Discover: admin,edit,view
+    Run-Delete: admin,edit,view
```
Comment (piyush-garg) [inline environment/templates/fabric8-tenant-jenkins.yml]: make this only admin

File: environment/templates/fabric8-tenant-jenkins.yml [hunk 3]
```diff
@@ -834,11 +834,44 @@ objects:
...
+    Run-Update: admin,edit,view
```
Comment (piyush-garg) [inline environment/templates/fabric8-tenant-jenkins.yml]: make also this only admin

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=6dff8e4cc5b89e84eb1b53c6c69c791dc79a41b2) and click the update tenant button

File: environment/templates/fabric8-tenant-jenkins.yml [hunk 4]
```diff
@@ -834,11 +834,46 @@ objects:
...
+    Credentials-View: admin,edit,view
+    SCM-Tag: admin,edit,view
+    Slave-Configure: admin
+    Slave-Delete: admin
 parameters:
 - name: USER_NAME
   value: developer
 - name: JENKINS_OPENSHIFT_VERSION
-  value: vb355188
+  value: SNAPSHOT-PR-234-140
```
Comment (hrishin) [inline environment/templates/fabric8-tenant-jenkins.yml]: Do we need to update it before merge?

Comment (piyush-garg) [reply environment/templates/fabric8-tenant-jenkins.yml]: Yes, we need to first merge PR on config repo. That will send an automated PR for image tag update after that we need to remove this part and merge.

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=00337e39bd23091fe91b43f56b4e66508aaf771d) and click the update tenant button

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=4e783d9effd3b087fac0b903faf9bbf1a5158363) and click the update tenant button

**Comment by piyush-garg** [comment]
[test]

**Comment by fabric8cd** [comment]
This PR changes one or more of the template files. These changes are now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?templatesRepoBlob=4e783d9effd3b087fac0b903faf9bbf1a5158363) and click the update tenant button

**Pull request merged**